core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -315,7 +311,16 @@ open class TileInfo {
                 }
                 stats.add(unique.stats.times(numberOfBonuses.toFloat()))
             }
-
+        
+        for (unique in observingCiv.getMatchingUniques("Tile yield from [] +[]%")) 
```

Comment:
If we're rewriting this, I would go with "+[]% yield from []", but even that's problematic.
You see, you're checking the unique filter, but the stats are for the improvement only, yes? So if it says like "+[50]% Tile yield from [Ocean]" it will still give the yield bonus to the improvement...

Reply:
I have implemented this, but by doing so also cleaned up some other code which _might_ result in incompatible mods, but I don't expect it. If you do expect it, please tell me and I'll revert the change. Change is mostly the split of the three uniques inside the ```if (city != null)``` block in this same function.